lib/pbench/test/unit/server/database/test_datasets_db.py

```diff
@@ -32,7 +32,8 @@ def test_construct(self, db_session, create_user):
         assert ds.name == "fio"
         assert ds.state == States.UPLOADING
         assert ds.md5 is None
-        assert ds.created <= ds.transition
+        assert ds.created is None
+        assert ds.uploaded <= ds.transition
```

Comment:
We create the `Dataset` when we start reading the file, but we can't read the `created` date until we have the complete tarball so we can open it and pull `metadata.log`. And I prefer to have it "undefined" rather than faking another date that could be misleading elsewhere. Essentially, a `Dataset` in `UPLOADING` state doesn't yet know when the dataset was created; that's life.